arangod/Replication2/ReplicatedLog/LogLeader.cpp

```diff
@@ -298,6 +298,7 @@ void replicated_log::LogLeader::executeAppendEntriesRequests(
                                            ResolvedPromiseSet> {
                           auto guarded = self->acquireMutex();
                           if (!guarded->_didResign) {
+                            // TODO: This can throw when we register the Callback
```

Comment:
Note to Reviewer: I created a Followup card for this TODO.
I think it is okay, but want to talk to the team about this next year.